### PR TITLE
benchmark: compare qa pipelines

### PR DESCRIPTION
**Pull request by gaurav274**
👋 Thanks for submitting a Pull Request to EVA DB!

🙌 We want to make contributing to EVA DB as easy and transparent as possible. Here are a few tips to get you started:

- 🔍 Search existing EVA DB [PRs](https://github.com/georgia-tech-db/eva/pulls) to see if a similar PR already exists.
- 🔗 Link this PR to a EVA DB [issue](https://github.com/georgia-tech-db/eva/issues) to help us understand what bug fix or feature is being implemented.
- 📈 Provide before and after profiling results to help us quantify the improvement your PR provides (if applicable).

👉 Please see our ✅ [Contributing Guide](https://evadb.readthedocs.io/en/stable/source/contribute/index.html) for more details.

### CODE REVIEW

File: apps/story_qa/langchain_qa.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+text_splitter = RecursiveCharacterTextSplitter(chunk_size=500, chunk_overlap=50)
+texts = text_splitter.split_text(open(book_path).read())
+
+# create FAISS vector index based on HuggingFace Embeddings
+print("Building FAISS Vector Index ...")
+ss = time.perf_counter()
+store = FAISS.from_texts(
+    texts,
```
Comment (jiashenC) [inline apps/story_qa/langchain_qa.py:88]: One minor thing if you want to have separate feature extraction and index creation steps with timing.

**Comment by jiashenC** [comment]
A bit messy here (https://github.com/georgia-tech-db/eva/compare/master...benchmark-qa-v2). Do you want to incorporate some fixes from this branch as well as part of this PR?

File: apps/story_qa/langchain_qa.py (hunk 1 above)
Comment (gaurav274) [reply apps/story_qa/langchain_qa.py:88]: It takes around `1.5s` to build the index in langchain, whereas we spend like `23s`. I'm going to profile what is going on here. 
Possible reasons: 
1. Low batch size
2. Sqlalchemy overhead in create table.

Comment (xzdandy) [reply apps/story_qa/langchain_qa.py:93]: We are creating a secondary table if I recall correctly. That may also lead to extra overhead.

Comment (gaurav274) [reply apps/story_qa/langchain_qa.py:93]: That is part of the `create_vector_index` call, which is actually fast.

Comment (gaurav274) [reply apps/story_qa/langchain_qa.py:93]: Create table  12919.144 ms
Extract features 28384.113 ms
Create index 326.020 ms

Comment (gaurav274) [reply apps/story_qa/langchain_qa.py:93]: @jiashenC Please verify the numbers for `langchain`

Comment (xzdandy) [reply apps/story_qa/langchain_qa.py:93]: Are these numbers from EVA with ray enabled? Could we get also numbers with ray off?

**Comment by jarulraj** [comment]
This PR subsumes #829

File: apps/story_qa/langchain_qa.py (hunk 1 above)
Comment (gaurav274) [reply apps/story_qa/langchain_qa.py:93]: Yes
Without ray:
Create table 13136.788 ms 
Extract features 10273.294 ms
Create index  282.686 ms

**Pull request closed without merging**